### DISCUSSION TITLE
システム改定の要望スレ（含Playground／サイト）

### DISCUSSION THREAD

**Discussion by snext1220** [snext1220/stext#265] [Ideas]
本スレでは、システムに対するリクエストを募集いたします。「こんな機能があればよい」という方は、リクエスト戴ければ幸いです。SText本体に限らず、Playground、サイトについてのリクエストも歓迎です。

#### 要望フォーマット

以下、軽いフォーマットです（沿って戴くと、把握が容易になるので助かりますが、必須ではありません）。

+ リクエスト概要（極力短く、端的にお願いできますと）
+ 用途（具体的な利用局面など。汎用性の有無判定用）
+ 仕様（`<scene hoge="xxxx">`のように寸例的に表して戴くと良いかと。設定可能な値なども）
+ UIイメージ（既存UIに影響する場合は、PPTなどでイメージを）
+ その他補足（あれば）

#### リクエストの扱い

戴いたリクエストについては、それぞれ別個にIssueを立てて、次回のUpdateで議論の上、対応の是非を決めていきたいと思います。対応の是非は、

+ システム的な可否
+ システム影響の大小
+ 他の皆さまの賛否

によって決めていければと思いますので、是非皆さまのご意見をお願いいたします。

※ただし、以下の条件を満たすものは、管理人判断で即反映させて戴きますのでご了承ください。

+ 既存機能に影響が出ない
+ 純粋な機能向上
+ システムへの影響範囲は軽微

ということで、本Issueはあくまでリクエスト請けのみの用途なので、詳細の議論は別Issue立てまでお待ちください（軽いリクエスト内容の確認のみ、こちらのスレで行わせていただく場合はあり）。

**Comment by Salvadors-cabin** [comment]
一件応募させていただきます。

・リクエスト概要
　「BattleSheetの機能拡張」です。

・用途
　戦闘システムの自由度を拡張することを目的としています

・仕様
　下の画像の①～⑥の機能を変更にしてほしいという要望です。

![BattleSheet2](https://user-images.githubusercontent.com/42834363/73122363-698b4c80-3fc7-11ea-8872-d60fa619e2f4.png)

①表示内容をカスタマイズ可能に。
②Free欄の数値を表示。ボタン押下時に更新、再表示。
③攻撃アイコンをクリックしたときに計算式実行。（例：敵のHP(Free欄)を減算）
④表示スペース縮小の為に計算式表示はやめ、「盾」のようなアイコンに変更。（計算機能は今のまま）
⑤戦闘ルールの文章表示。
⑥イラスト表示。

データの方は、enemyカテゴリに追加してはどうかと思います。
以下のような例です。
<enemy id="m01" name="こうもり" element="" attack="" func="" drop="" ①bar="名前／属性　　HP" ②value="free1" ③damage="free1" func1="L+R+FREE3" ⑤rule="特別ルールhogehoge" ⑥image="hoge.png">

※互換性維持のため、①～⑥を設定しない場合は従来表示。
※①⑤⑥は複数の敵が同時に出現する場合は最初の敵のを有効。

これはもともと「ラミディア」の戦闘システム自動化を考えていたときに思いついたアイデアですが、自由度が広がれば他のシナリオでも利用できると思います。

・UIイメージ
　「ラミディア」を例にすると以下のようなイメージになると考えています。

![BattleSheet3](https://user-images.githubusercontent.com/42834363/73122368-7314b480-3fc7-11ea-89d3-8078ef311e74.png)

※複数敵が登場するとFree欄が敵の数だけ必要になります。ラミディアでは問題無いですが、将来的にはFree欄の拡張も必要になるかもしれません。

検討いただけたら幸いです。

**Comment by toki-sor1** [comment]
【1/30追記】
これだとサイトの方のリクエストですね…！失礼しました;
SText自体のリクエストでは、今のところ汎用性のあるものは、なかなか思いつかないので、今回は保留とさせてください;
***

リクエストが遅くなり、申し訳ありません…！

STextはシナリオへの反応が、プレイヤー側・作者側ともに分かり辛く、ゲームが遊べるサイトにもかかわらず、関係者以外の気配がないのも課題かなと思いました。

> リクエスト概要

「シナリオ紹介ページに投票フォーム設置」

> 用途

- 紹介ツイートする、RTするといった反応はハードルが高い。シナリオへの反応・応援のハードルを下げる目的
- プレイヤーさんに対し、「他のプレイヤーさんの存在」を明確にする
- 投票項目に「ストーリー」「キャラクター」など設定することで、シナリオ内でどの要素が好まれているのか分析する

> UIイメージ

- Twitterのような投票フォームをシナリオ紹介ページ下に設置（画像参照）

![image](https://user-images.githubusercontent.com/42646342/73365062-ffd4b080-42ee-11ea-8b06-243b70b6f51e.png)

**Comment by cocotori** [comment]
playgroundを使っていて感じたことなのですが、
未保存でページを離れることに対する警告が出せたらいいなと思いました。

こまめな保存に気をつければいい話なのですが、
ついうっかり保存する前にブラウザを落としちゃった、
という事態もどうしても発生すると思うので。

ブラウザに自動で一時保存できれば一番いいのかもしれませんが、
そこまでできなくても警告が出るだけでもいいかと思います。

**Comment by Salvadors-cabin** [comment]
リクエストさせていただきますー。

・リクエスト概要
　「Characterシートの機能拡張」

・用途
　戦士系、魔法系で有利不利（またはストーリー変化）のあるシナリオをユーザの選択でプレイできるようにするため、ゲーム開始時にプレイヤーに性別／クラス、AGEを選択できるようにする。
　また、キャラ名に好みの名前を付ける事で、より感情移入できるのではないかと思います。
　※キャラ名を入力した場合はニックネームは無しにする。

・仕様
　ゲーム開始時に表示されるCharactorシートで、キャラ名の入力、性別／クラス、AGEの選択を可能にする。
　※現在選択できるのは職業だけだが、これを拡張するイメージ。

ご検討いただけたら幸いです。

**Comment by snext1220** [comment]
PgFlow周りのリクエストをちらちら伺っているので、メモ代わりにこちらにまとめておきます（随時更新）。ある程度落ち着いたところで、別Issueを立てたいと思います。

+ ~PgFlowの機能を整理したい（https://github.com/snext1220/stext/issues/207#issuecomment-663854455 ）~
+ ~シーンのカラーを設定したい（http://chanagame.hateblo.jp/entry/2020/07/25/190211）~
+ target属性がないアイテムでも個数を利用可能に（ https://github.com/snext1220/stext/issues/295#issuecomment-891354570 ）
+ リンクをまとめてコピーしたい （https://github.com/snext1220/stext/issues/297#issuecomment-894758265 ）

2020/08/02追加
以下、snext1220/snext-management#18 (comment) 由来ですが、主にライブラリの制約ですぐさまに対応が困難なものです。ただし、戴いたご意見はご尤ものため、今後の長期的改善のために、ここに残しておきます。

+ リンクを削除する動作はundoできない？
+ チャート上でシーンを選択した際、そのシーンにつながる矢印が色づけされるのはとても見やすくて良い。欲を言えば、つながっているシーンも色が変わるともっと良い。あと矢印に示される選択肢の内容が重なって表示されるので、ずらしてもらえると。（欲張りすぎ？）

**Comment by RYU-DS** [comment]
どうしても必要という程ではないのですが、シーンでHPなどを増減させる際に特定の数値を指定できるようになりませんでしょうか？
というのも、幕末ソーサリアンでのキャラ引継でこれらの数値を前シナから持って来てしまうわけですが、シナリオごとに用途の違うFREEはいったんリセットする必要があるので、できれば次シナ開始時に0にしてしまいたいのです。
現状ではステ強化シーンでプレイヤーに手動で0にしてもらう事で対応しようと思っていますが・・・

**Reply by toki-sor1** [reply]
もしかしたら、下記の手順で代用できるかもしれません。

例）scene0（プロローグ）でfree1の数値を0にする場合

scene0のfree1に固定値として「@0」と設定すると、数値をリセットできるかなと思います〜。
（固定値なので、設定したそのままの数値になります）
***
なるほど〜、今のところfree1〜3の数値を引き継ぐ必要のあるシナリオがないので、シナリオが揃った段階で、各シナリオでもfree1〜3の数値を最初にリセットする設定に変更するのも良いかもですね。
（要相談になるかなと思いますが）

ご意見ありがとうございました！

**Reply by RYU-DS** [reply]
コマンドにあったんですね失礼しました・・・
で、どこにそれが書いてあるのかとPGflowのFREEをWクリックしてみたんですが・・・
1回だけ「@での指定値ができる説明」が出た後、以後何回クリックしても画像の通り「状態異常付与に関する説明」しか出ず・・・
実際今までも何度もクリックしましたが「状態異常付与に関する説明」しか出た事が無かったのでビックリしました。
なんでしょうかこれ・・・
![スクリーンショット 2021-11-05 013517](https://user-images.githubusercontent.com/43634197/140382432-ecf37462-52ab-4c6f-a168-83693f06355c.png)

**Reply by snext1220** [reply]
上記件失礼致しました。
ページ内の`</label>`閉じタグのヌケが原因でした；
先ほど修正し、本番環境にも反映致しましたので、キャッシュクリアの上でお試しくださいm(_ _)m

**Reply by RYU-DS** [reply]
確認しましたー
対応ありがとうございます。
![スクリーンショット 2021-11-05 215110](https://user-images.githubusercontent.com/43634197/140513044-31617f1f-b37f-4c8c-9209-11b8f6ef79ac.png)

**Comment by RYU-DS** [comment]
おそらく難しいだろうとは思うものの、一応希望として挙げておきたいのですが…

魔法の効果を指定・変更できるようになりませんでしょうか？

実例としてはやはりHEALの回復量の変更でしょうか。
また今回の龍馬ではFREEに味方HPを設定し、ソーサリアンのHEAL的には味方も回復させたい所なのですが…
すでに味方への攻撃のための欄が多く、さらに味方回復のための欄を増やすのは断念した次第です。
これをHEALの効果を「HP・FREE1・FREE2を+20」などとできれば色々と幅が広がると思います。

自動反映のON/OFF、効果のテキストの変更だけでもできれば大変助かるのですが…
![スクリーンショット 2021-11-05 222047](https://user-images.githubusercontent.com/43634197/140517817-226f22e7-88ce-43a4-8ad7-b11ad3e52eda.png)

**Reply by snext1220** [reply]
ご提案ありがとうございます！
魔法のカスタマイズ、魅力的ではあるのですが、ご推察の通り、システムのコアにも影響することから少なくともすぐさまの対応は難しく、別途検討とさせてください（現在、システム全体としてのカスタマイズ自由化も考えており、長期的には魔法システムをモジュール切り出すなどしていきたいと考えています）。

**Reply by RYU-DS** [reply]
ですよねー
しかしなんとか自動反映のOFFだけでもできませんでしょうか？
そのスレでMAX半分回復を手動でする際に15引いてくれとか煩雑過ぎて・・・
15ぶんはボーナス扱いとかでもいいんですが・・・

**Reply by RYU-DS** [reply]
ついでになんですが、ボーナスアイテムの効果の自動反映もシナリオ側で解除できるようになりませんかね？

これに限らず、自動反映はどれも解除できるようになっていないと、シナリオ側の自由度がかなり制限されてしまうと思うのですが・・・

**Comment by RYU-DS** [comment]
以前から考えていたのですが（というか誰でも一度は考えた事があると思いますが）

ダイスの数値を好きな数値に指定する

を可能にできるといいのではないかと思いますがどうでしょうか？
ダイス増加の議論でも出てましたがインチキ能力・効果としての需要があるかと。
当方の実例としては特殊武器の効果や魔法REJUVENATEの拡大解釈効果として使用したいと思っています。

ダイスの片方をWクリックすると数値が1ずつ上がっていく

とかでどうでしょうか？

**Reply by toki-sor1** [reply]
@RYU-DS さん

確かに、現在のダイスの仕様では、手動で振るorウィンドウの開閉でランダムな数値にしかできませんものね。
狙った数値を出すことができれば、テストプレイ時にバトル関連のダメージ調整等もスムーズに行えそうに思います。

また、使用例も挙げていただきましたが、特殊武器の効果や魔法の拡大解釈効果というのも、シナリオの表現の幅が広がったり、より奥深い様々なバトルが楽しめるアイデアだと感じました！

**Comment by RYU-DS** [comment]
「既に生成されているランダムリンクのランダム移動先の追加」ができなくて困っていたところ
移動先をカンマで区切る基本方式でランダムリンクを新たに作り直す事で解決したのですが…

そうして作ったランダムリンクは編集する事ができず、編集するにはまた作り直す必要があるようです。
追加ができない事も含めて一応問題は無いのですが非常に不便なので報告まで

![スクリーンショット 2022-03-14 024314](https://user-images.githubusercontent.com/43634197/158072337-f7dd9f3f-983f-4066-8059-652aea5f4ce7.png)

**Reply by toki-sor1** [reply]
【23:35頃 追記】す、すみません……！
注意点の辺りに誤りがあり、訂正いたしました；

---
ご報告ありがとうございます～！
確かに、「既に生成されているランダムリンクのランダム移動先の追加」は、Playground Flowの「シーン」タブor「リンク」タブからだと追加できませんね；；

そこでRYUさんのご提案を参考に、解決策を考えてみました～（がたがたマーカー線になってしまい、どうもすみません；；）。

![image](https://user-images.githubusercontent.com/42646342/158191233-ec955279-3cda-40bd-842c-e43602955f94.png)

1. 念のために **「現在制作中のシナリオを保存」** する（保存は任意です）
2. Playground Flowの上部メニューから「Playground **Editor**」に切り替える
3. 「既に生成されたランダムリンク」を見つける（**Ctrl+Fキーの検索機能**でscene id="○○"と指定すると見つけやすいかも）
4. 「追加したいランダム移動先」を**カンマで区切って入力する**
5. 上部メニューから「Playground **Flow**」に戻る

**※FlowとEditorを行き来する際の注意点※**
　「FlowとEditorを行き来する際」に、**各Playgroundに移動した後は、移動前のページは暗く表示されます。**
　**現在編集中のページは、暗くならず正常に表示されている**ので、暗く表示されたページは、閉じていただいても大丈夫です。
　（**各Playgroundに移動する前にシナリオを保存しておく**と安心かもです！）

こちらはFlowとEditorを行き来したりするのが手間ですが、上記の検索機能と併用することで、「既に生成されているランダムリンクのランダム移動先の追加」もちょっと楽になるかなと思います～。

**Reply by cocotori** [reply]
おそらくですが、リンクの表示順序を設定することで
解決できるのではないかと思っています。

いろいろ試しているのですが、
同じひとまとめのランダムリンクは
表示順序を同じ番号にする必要があるようです。

逆に違う番号にすれば、複数のグループのランダムリンクを作ることもできるようです。
（1,2,3のランダムリンクと4,5,6のランダムリンクなど）

いずれにしても、「シーン追加」のボタンは
現状は通常リンクしか作れないので
一度通常リンクを作ってから、ランダムリンクに変更する手間はありますが。

ご参考になれば。

![クリップボード](https://user-images.githubusercontent.com/42885160/158587235-ba528c7c-90cb-42c5-86cb-a0497dc5949c.png)

**Reply by RYU-DS** [reply]
EditerでSGMLにして何とかするのは今までも何回かやってるのですが、自分的には「最後の手段」と思ってるので…

ランダムリンクの編集もそれならできるのは確認しましたが、ちょっと「？」となったのが…
Editerで編集したら、編集できないカンマ区切り表示（画像１）が、編集できる複数リンク表示（画像２）に勝手になりました。
画像１
![スクリーンショット 2022-03-16 223530](https://user-images.githubusercontent.com/43634197/158606246-5f6a2a35-1088-4670-afab-17d72e2f0596.png)
画像２
![スクリーンショット 2022-03-16 222317](https://user-images.githubusercontent.com/43634197/158606356-bf56d928-f106-4758-a8ba-ba14d0ce7f26.png)

しかし行き先の追加は相変わらずできず、追加行先リンクをランダムとして作ると他の行き先の入力を求められるので入力するともう一つランダムリンクができ、追加行先リンクを通常リンクとして作ってランダムに変更して表示順序を合わせても別のリンクとして表示されるのに変わりはありませんでした。（画像３）
画像３
![スクリーンショット 2022-03-16 225840](https://user-images.githubusercontent.com/43634197/158606869-9db662c0-f552-4e16-9606-64306c6dba52.png)

現状、Editerに移動するかして移動先をカンマ区切り入力するしか無いようですね。
あとFlowの方で移動先をカンマ区切り入力する所が「from[シーンNo.]～to[シーンNo.]」の所なのが
説明ではちょっとわかりにくいと思います（少なくとも自分にはわかりませんでした）。
![スクリーンショット 2022-03-16 230828](https://user-images.githubusercontent.com/43634197/158609436-ea5ef389-4d83-4e9b-88a6-2de32a7e5eb3.png)
![スクリーンショット 2022-03-16 222544](https://user-images.githubusercontent.com/43634197/158609460-ac3324d3-a9ff-4abb-8e36-cd91e374bd55.png)

**Reply by cocotori** [reply]
＞RYUさん
　
間違った対策法を伝えてしまい、すみませんでした。
　
　
＞みなさま

こちらの問題、なんとなく原因が見えてきましたので共有させてください。
今回の問題とissueの方で挙げさせて頂いた問題は
根っこが同じだと思います。

・「すでに作成されたランダムリンクにランダム移動先の追加できない」
・「複数の条件式リンクが、一つにまとめられてしまう」

（ランダムリンクが作りにくい、説明がわかりにくい等は別問題として
　議題とさせていただければ）
　
これらの問題は、

・「xml形式とjson形式（playground flowのデータ方式）の変換時の齟齬」
が問題になっているようです。

例として、以下のランダムリンクを作ります。

![図1](https://user-images.githubusercontent.com/42885160/159110720-6632aa1b-e95a-4b68-a794-ace4b61d3d02.png)

これを、

　1) flowでリンクを作り、そのままflowでjson形式で保存した場合

　2) editorでリンクを作り、flowに移動してjson形式で保存した場合
　3) editorでリンクを作り、xml保存してから、flowで読み出してjson形式で保存した場合

をそれぞれ行います。1)は、
![図2](https://user-images.githubusercontent.com/42885160/159110877-a3891903-1333-444a-83aa-184a91f7081c.png)

　2),3)は、
![図3](https://user-images.githubusercontent.com/42885160/159110764-2bfdbb34-65c0-4662-b34b-9d8df9d6f12b.png)

のようになります。
一度でもxml形式を経由すると、リンクの順番（order）の「””」が取れてしまうようです。

この状態で新しいランダムリンクを追加した例が下の図です。
![図4](https://user-images.githubusercontent.com/42885160/159110939-55e41e22-d13f-4d7c-ab70-3f2802eb33f8.png)

数字は同じですが、これが別扱いとなって、別リンクになってしまいます。

具体的に問題になるのは、
・過去の公開作品のxmlファイルをflowで読み込んで修正する
・flowで作っていたものを、editorに移動して確認や修正を行い、またflowに戻す

などです。現状では極力flow一本で作っていただくのと、
json形式でも最新データを残しておくのが安全かと思います。

**Reply by RYU-DS** [reply]
調査お疲れ様です。

＞間違った対策法～
既に試していた事を書くのをサボった私が悪いだけなので気になさらずに。

なるほどー
こうやって原因がわかるとスッキリしますね。
そのスレはGBATで作ってSGML化してUPしたものをFlowに持ってきてるので、もうどうしようもないという事ですねｗ

言ってる通り現状どうとでもできるので問題は無いのですが、こうやって報告するのはFlowで完結できることを目指していると解釈しているからなので、もしEditerとの併用が前提ならば改良も必要無いかと思います。
ただ「Flow側でこういう事をしようとすると問題が起きる」という注意書きは必要かと。

**Comment by RYU-DS** [comment]
以前から少し問題に思っていたのですが、いよいよ厳しくなってきたので提案させていただきたいのですが…

Battle Sheetが他のウィンドウを開くと閉じてしまう

のは何とかならないものでしょうか？
当方のシナではBattle Sheetが閉じるたびにダイスの数値が変わってしまうのが大問題で、戦闘の戦略性を維持しつつ少しでも簡略化するために一度出た数値で同様に～としようにも、手動でのHPMP増減どころか魔法を使っても数値が変わってしまいます。

そもそも敵と戦闘中であるのにそれに必要な行動を取っただけで、Battle Sheetが閉じていったん戦闘が中断されたような感じになるのは、戦闘の臨場感を大きく損ねていると思います。

ただこれはUI変更時にも少々あった話で、stext全体から見た戦闘の重要度の問題でもあり、それの蒸し返しになってしまうのですが…この状態で色々と試行錯誤し、テストプレイを重ねた上での私の感想として再提案させて頂きたいと思います。

**Reply by toki-sor1** [reply]
@RYU-DS さん

貴重なご意見＆ご提案ありがとうございます！

> Battle Sheetが他のウィンドウを開くと閉じてしまう

確かに、現状の仕様だとBattle Sheet展開時に魔法やアイテムの使用等で他のウィンドウが開くと、その後にBattle Sheetを開き直した際にダイスの数値が変わってしまいますね；

STextデフォルトの戦闘ルールでは、ダイスを振るのは基本的にダメージ式ボタンを押す前＆魔法を使用するのはターン開始前（という認識）なので、Battle Sheet開閉でダイスの数値が変わっても不自由は感じていなかったのですが、**シナリオ毎の特色ある戦闘ルールが制限されてしまう仕様でもあると思います。**

仰る通り、戦闘中に他のウィンドウを開くと、戦闘が中断された形になる＆ダイスの数値が変わってしまうのは、臨場感を損ねている要因にもなっていると思います。
ただ、STextはスマホといった小さな携帯端末からも遊べる仕様上、ウィンドウの複数表示は画面の大きさ的に厳しそうなので、 

- **BattleSheetの上部または下部に、折り畳み式でMagic&Stars等を格納する**
- **Battle Sheetの開閉でダイスの数値が変わらない仕様をデフォルトにする**

……のような感じにできればいいですよね。
プログラムは専門外なので、私からは問題が解決可能かどうかお答えできなくてすみません；
ご担当のリオさんからのご回答をお待ちくださいませ～。

**Comment by RYU-DS** [comment]
itemの各設定に関してもいくつか希望がありますので重要度(希望度)が大きいものから順に…

・itemの個数の設定をドロップでも設定できるように
そのスレではドロップがメインなので、現在同効果の別名アイテムを複数設定してる有様です。
※際限なく取れてしまうので個数の上限の設定もできれば。

・itemの効果を複数設定できるように
scene要素が使える的な名目なので入れ忘れかもしれませんが…
HPMP両方回復くらいはしたい所で、現状HP回復は自動だがMPは手動で回復とか変な事になってます。

・(scene要素も含め)HPなどの「halfに固定」以外に「half分を+/-」も欲しい
現在値がMAXとは限らない以上、「halfに固定」は増える事も減る事もあるというのは変で、非常に使いづらいです。
「○○に固定」で○○が指定できれば使い道は多いかと。

・USEしても無くならない設定も可能に
これはドラクエなどの「使うと効果がある武器」を設定するのが主な使い道で、現状でも効果の説明書きだけして手動で行ってもらう事はできますが、HP回復など自動化できるものや使用時に「剣から冷気がほとばしる！(FREEZEの効果)」と表示されるだけでも臨場感がだいぶ違うと思います。
また剣にSTR+10と設定しておけば「USEで装備」とする事もできます(外す時は手動で-10ですが)。

・敵からの攻撃要素に「七星欠片を減らす(指定/ランダム)」があると面白そう
おまけですが、特定の魔法を使わせないように狙えるのは多分どんなシナでも使えそうかと。

**Reply by toki-sor1** [reply]
@RYU-DS さん

あわわ、本件について確認が不十分でご返信が遅くなり過ぎてしまい、大変申し訳ありません……！！
既に解決済みの問題もあるかもしれませんが、私でお答えできる範囲で回答させていただきます！

> ・itemの個数の設定をドロップでも設定できるように

こちらについては**敵タブの「ドロップ」欄に「item/i01:@10/毒消し」のようにご入力していただけますと、itemの個数も指定できました（ドロップボタンには個数が表示されません）。**

> ※際限なく取れてしまうので個数の上限の設定もできれば。

上記の方法で「毒消しｘ10個（item/i01:@10/毒消し）」と設定した場合、
**フルで10個所持している時にドロップボタンを押すと10個以上にはならなかった**のですが、**9個の状態でドロップボタンを押すと補充されて10個になってしまいますね。**

ただ、**敵からitemをドロップしない限り＆Sceneで該当itemを拾わない限りは、指定の個数以上には増えないので**（毒消し10個なら10個以上にはならない）、シナリオの設定次第では許容範囲かなと思います。

※ただ、例外的に**最初に毒消しを1個所持している状態で、敵の「毒消しｘ10個（item/i01:@10/毒消し）」と設定したドロップボタンを押した場合、11個とはならず10個に上書きされてしまう**ようなので、注意が必要だと感じました。
***
> ・itemの効果を複数設定できるように

こちらもPlayground Editorに切り替えて「HP,MP」と指定してみましたが、上手く反映されませんでした。
**HPMPの同時回復アイテムの実装は、現時点では難しそうです；；**

***
> ・(scene要素も含め)HPなどの「halfに固定」以外に「half分を+/-」も欲しい
・USEしても無くならない設定も可能に

>・敵からの攻撃要素に「七星欠片を減らす(指定/ランダム)」があると面白そう

現状では上から順に、代替案がなかったり、手動に頼る又はSceneのイベント演出で実装するといった方法しか（おそらく）ないのですが、**上記3つのご提案が実装されると、遊び易くなるのは勿論のこと、よりコンピューターゲームの雰囲気が増して面白そうですし、STextの独自性も更に強化されて良さそうですよね！**

**Reply by RYU-DS** [reply]
ドロップitemの個数設定できたんですね…orz
でも累積できないんじゃちょっと意味無いかな～と思ったんですが、シーン要素の方のitem数パラがリロードで累積されるのを見たので、もしかしてと試してみた所…
@の個数指定でなくi01:10ように増加数指定だけにすればどんどん累積する事が確認できました！
あとはi01:1とすれば取るごとに1個ずつ増えていく薬草などができます。
なおやはり上限は無いようです（とりあえず100は超えましたｗ
![スクリーンショット 2023-09-23 041003](https://github.com/snext1220/stext/assets/43634197/e4c023ea-416a-417f-adab-ab2cb9cca67e)

**Comment by RYU-DS** [comment]
itemの複数設定でちょっと困っています。

itemの名前と説明が長くてウィンドウに収まらない場合はDETAILSで見ればいいのですが…
item個数の×nがDETAILSでは表示されず、右へスクロールやウィンドウ拡大などもできないため、現状では残り個数を確認する手段がありません。
×nをitem名の前後などにするか、DETAILSでも表示するような改善が必要かと思います。

画像では「どくがのこな」を複数持っていても残数を確認できず、「タリスマン(大)」は残り使用回数を見えるようにしてありますがかなり無理をしている状態で、全角で名前＋効果をこの文字数に必ず収めるのは当シナリオに限らず厳しいのではないかと思うのですが…

![スクリーンショット 2023-09-17 233529](https://github.com/snext1220/stext/assets/43634197/c4a8a4d0-7bc0-41a1-aa2f-f08270847293)

**Reply by toki-sor1** [reply]
@RYU-DS さん

シナリオの更新お疲れ様です！
ご意見＆ご提案ありがとうございます！　こちらの問題についても拝見いたしました！
システムのプログラム的なことについては、私自身の知識不足で正確な回答になっていないかもしれませんが（大汗）、私自身が試したことに基づいて回答いたしますね。

itemの名前や説明文が長い場合、確かに個数を確認できないのが現状ですね……。
itemの説明文がアイテム欄で改行できるかどうか、半角スペース2つ、brも入力してみましたが、結果は以下のとおりです。

+ 半角スペース2つの場合は**改行されなかった**
+ brの場合は**DETAILSで表示される説明文のみ改行されていた**

ご指摘のとおり、**DERAILSでは個数が表示されないため、問題解決には至りませんでした。**

説明文については、やはり元々が**15文字程度を推奨**となっているため、基本的に長文の説明文は想定されていないように思います。
ただ、アイテムの効能によっては、どうしても説明文を簡略化できないケースもありますよね……。
そうなると現状では、**説明文の分かり易さを取るか、それともアイテムの残数表示を取るかの2択になってしまう**のかな、と思います；；

（説明文の最初に使用回数を表示する、という方法も考えてみましたが、アイテムの使用回数とアイテムの残数は意味が異なるため、ボツになりました……。
また、式埋め込みも試したものの、アイテム名と説明文しか表示されず、こちらもボツに……）

この度は、良い代替案がご案内できず申し訳ありませんでした……！
**ご提案いただいた「×nをitem名の前後などにするか、DETAILSでも表示する」という表示方法になれば、表現の幅が広がりますし、とても見やすくて良いと思います～！**

**Reply by RYU-DS** [reply]
私の方でも簡略化もしくは「現在の残り使用回数」ではなく「n回使うと無くなるという意味の注意書き」でなんとか対応かなーと思っています。
